Branch 16 add 自動テストfor converter

L2206693309 opened the pull request:
コンバーターの自動テスト作成

L2206693309 committed:
26回コミット add コントローラーの自動テストメソッド追加
自動テスト追加(更新,登録処理を除く)

L2206693309 committed:
27回コミット add コンバーターの自動テストメソッド追加
自動テスト追加

KeitaUenishi commented on src/main/java/raisetech/StudentManagement/controller/StudentController.java:
```diff
@@ -115,7 +115,7 @@ public ResponseEntity<StudentDetail> registerStudent(
   @Operation(summary = "受講生更新", description = "受講生を更新します。")
   @PutMapping("/updateStudent")
-  public ResponseEntity<String> updateStudent(@RequestBody StudentDetail studentDetail) {
+  public ResponseEntity<String> student(@RequestBody StudentDetail studentDetail) {
```
こちらのメソッド名は特に変更不要かと思いましたが、`student` にされた意図としては何でしょう？

L2206693309 replied:
意図しない変更だったようです。戻しておきます。

KeitaUenishi commented on src/test/java/raisetech/StudentManagement/controller/StudentControllerTest.java:
```diff
@@ -0,0 +1,154 @@
+
+  @Test
+  void 受講生詳細の受講生でIDに数字以外を用いた時に入力チェックに掛かること() {
+//  Students の id を Integer で宣言しているため実装不可
+
+/*
+    Students student = new Students();
+
+    student.setId("テストです。");
+    student.setName("角田 憲哉");
+    student.setFurigana("Kakuta Kazuya");
+    student.setNickname("Kazuya");
+    student.setMailAddress("l2206693309as@gmail.com");
+    student.setLivingArea("日本-神奈川県");
+    student.setGender("man");
+
+    Set<ConstraintViolation<Students>> violations = validator.validate(student);
+
+    assertThat(violations.size()).isEqualTo(1);
+    assertThat(violations.size()).extracting("message").containsOnly("数字のみ入力するようにしてください。");
+ */
```
ここのコメントがどう言ったことかわからなかったです。
バリデーションのテストだと思いますが、テストが実行できないということでしょうか？

L2206693309 replied:
Students クラス の変数 id を Integer で宣言していました。なので、Integer型で宣言されたidにStringを代入する必要のあるこのテストは実行できません。

KeitaUenishi replied:
確かにオブジェクトに文字列を代入しようとするとできないですね。
この場合だと登録処理のテストなどのように、リクエスト送信時のURLのパラメータを文字列として送信した際にどンなレスポンスが返却されるかを検証する、という方法であればテストはできると思います！

L2206693309 replied:
わかりました。リクエスト時のパラメータとしてidに文字列を指定しテストします。実行結果はこのPRのコメントに追記します。その際、Discordにてお知らせします。
レスポンスについては、ステータスが400であることを確認できれば大丈夫ですか？

KeitaUenishi replied:
> レスポンスについては、ステータスが400であることを確認できれば大丈夫ですか？

そうですね！ そちらで問題ないです！

KeitaUenishi commented on src/test/java/raisetech/StudentManagement/controller/converter/StudentConverterTest.java:
```diff
@@ -0,0 +1,39 @@
+    tester = new StudentConverter();
+  }
+
+  @Test
+  void コンバーターで変換処理ができること() throws Exception {
+
+    List<Students> studentsList = new ArrayList<>();
+    List<StudentCourses> studentCoursesList = new ArrayList<>();
+    List<StudentDetail> studentDetails = new ArrayList<>();
+    if (studentDetails.equals(tester.convertStudentDetails(studentsList, studentCoursesList))){
+      
+    }else {
+      throw Exception ;
+    }
+    
+  }
+}
```
テストにはなっていないと思いましたが、実装途中でしょうか？

KeitaUenishi commented on src/test/java/raisetech/StudentManagement/controller/converter/StudentConverterTest.java:
```diff
@@ -0,0 +1,39 @@
+
+  @Mock
+  private StudentConverter tester;
+  private java.lang.Exception Exception;
```
こちらはimportでExceptionを宣言してしまいましょう

L2206693309 replied:
追加します。

KeitaUenishi commented on src/test/java/raisetech/StudentManagement/controller/StudentControllerTest.java:
```diff
@@ -0,0 +1,154 @@
+  }
+
+  @Test
+  void 受講生登録ができること() throws Exception{
```
受講生登録のなにが実行できるか、まで具体的な方がいいですね
例：受講生登録が正常に実行できること

L2206693309 replied:
メソッド名を変更します。

KeitaUenishi commented on src/test/java/raisetech/StudentManagement/controller/StudentControllerTest.java:
```diff
@@ -0,0 +1,154 @@
+  }
+
+  @Test
+  void 受講生更新ができること() throws Exception{
```
こちらも登録のテストと同じく、もう少し具体的なメソッド名である方が良いです！

L2206693309 replied:
登録と同じく、メソッド名を変更します。

L2206693309 committed:
28回コミット update コンバーターの自動テストメソッドやコントローラーのテストの修正
コンバーターの自動テスト完成
コントローラーの自動テストのメソッド名およびテストの修正
コントローラーのupdateStudentが誤ってstudentに変わっていた問題を修正

L2206693309 commented:
コンバーターとコントローラーのテストクラスのテスト結果
![image](https://github.com/user-attachments/assets/a0d85cbc-bf93-4ff6-b3f2-1accc469355d)
![image](https://github.com/user-attachments/assets/2473693a-e4c8-49e8-bcfb-210c5e7a6c4f)
